synapse/storage/databases/main/registration.py

```diff
@@ -191,11 +191,22 @@ def get_user_by_id_txn(txn: LoggingTransaction) -> Optional[Dict[str, Any]]:
 
             return rows[0]
 
-        return await self.db_pool.runInteraction(
+        row = await self.db_pool.runInteraction(
             desc="get_user_by_id",
             func=get_user_by_id_txn,
         )
 
+        if row is not None:
+            # If we're using SQLite our boolean values will be integers. Because we
+            # present some of this data as is to e.g. server admins via REST APIs, we
+            # want to make sure we're returning the right type of data.
+            boolean_columns = ["admin", "deactivated", "shadow_banned", "approved"]
+            for column in boolean_columns:
+                if not isinstance(row[column], bool):
+                    row[column] = bool(row[column])
```

Comment:
slightly worrying as `bool(None) is False`. I think this is OK now because it's all coalesced, but seems like an easy way to trip up in future. Probably OK as I guess it will be applied consistently and a mistake will be noticed early on..?

Reply:
Yeah good point, adding a comment to try to catch this in the future.